### PR TITLE
TT-8833 MDCB visibility of edge gateways

### PR DESCRIPTION
**Pull request by tbuchaillot**
## Description

MDCB PR: https://github.com/TykTechnologies/tyk-sink/pull/356

## Related Issue

https://tyktech.atlassian.net/browse/TT-8833
## Motivation and Context

https://tyktech.atlassian.net/browse/TT-8833
## How This Has Been Tested

Added some unit tests - created https://tyktech.atlassian.net/browse/TT-9117 to have a better rpc testing in the gateway overall since at this moment it's a headache.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning why it's required
- [ ] I would like a code coverage CI quality gate exception and have explained why

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4830/merge
Commit:  
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4324763456)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4830/merge
Commit: b46b8f966de2927698c8d8052ee423ed59c3f89b 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5154505172)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4830/merge
Commit: d04235c944c888a578db47a7dd8a91b5e165d563 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5156264967)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4830/merge
Commit: 631121c5230ddaf2f46d9b7a6f6b182014c0ac68 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5156858745)

**Comment by Tyk-ITS** [comment]
**API tests result: cancelled** :no_entry_sign: 
Branch used: refs/pull/4830/merge
Commit: 849bae7823317cf51aaaa903b9f667f5aec866c6 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5156176905)

**Comment by Tyk-ITS** [comment]
**API tests result: cancelled** :no_entry_sign: 
Branch used: refs/pull/4830/merge
Commit: 8de37532767a3e663f94466a7b2ac7df906bae92 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5156268669)

**Comment by Tyk-ITS** [comment]
**API tests result: cancelled** :no_entry_sign: 
Branch used: refs/pull/4830/merge
Commit: f4a9b869a96e100ce8599d36e4bf6830b7f34f6f 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5156335739)

**Comment by Tyk-ITS** [comment]
**API tests result: cancelled** :no_entry_sign: 
Branch used: refs/pull/4830/merge
Commit: 104d29adb876683727369cf6d13cb30b9cf8d8bc 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5156816374)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4830/merge
Commit: 80b18d83c53914b0cb4c10dfcd7d697b3b20a693 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5187414951)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4830/merge
Commit: a3d3ebbf2cd18c8cb34b361854dc793f1111827f 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5187423583)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4830/merge
Commit: 5463bb6c1b46eefc9c15b3e69676468699c031ac 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5198275946)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/4830/merge
Commit: c0bc7e223a8afa09625fae088910d3597f978af7 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5198173159)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4830/merge
Commit: d43c5b8198195d5e5232653d95f863f089a334db 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5198118583)

File: apidef/oas/operation_test.go [hunk 1]
```diff
@@ -206,7 +206,7 @@ func TestOAS_RegexPaths(t *testing.T) {
 		_ = oas.getOperationID(tc.input, "GET")
 
 		pathKeys := make([]string, 0, len(oas.Paths))
-		for k, _ := range oas.Paths {
+		for k := range oas.Paths {
```
Comment (tbuchaillot) [inline apidef/oas/operation_test.go:209]: linter change

File: goplugin/mw_go_plugin_test.go [hunk 2]
```diff
@@ -478,7 +478,7 @@ func TestGoPluginMiddleware_ProcessRequest_ShouldFailWhenNotLoaded(t *testing.T)
 		v.UseExtendedPaths = true
 		v.ExtendedPaths = apidef.ExtendedPathsSet{
 			GoPlugin: []apidef.GoPluginMeta{
-				apidef.GoPluginMeta{
+				{
```
Comment (tbuchaillot) [inline goplugin/mw_go_plugin_test.go:481]: linter change

File: gateway/rpc_test.go [hunk 3]
```diff
@@ -4,60 +4,20 @@
...
+	"github.com/stretchr/testify/assert"
+
 	"github.com/TykTechnologies/gorpc"
 	"github.com/TykTechnologies/tyk/apidef"
 	"github.com/TykTechnologies/tyk/config"
 	"github.com/TykTechnologies/tyk/rpc"
 	"github.com/TykTechnologies/tyk/test"
 )
 
-func StartSlaveGw(connectionString string, groupId string) *Test {
```
Comment (tbuchaillot) [inline gateway/rpc_test.go:19]: moved to `testutil.go`

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4830/merge
Commit: 16dc07f543c7ce25b4248670d4d790f2bc5a93c8 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5198440425)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/4830/merge
Commit: f2a620ef25561d8ccee37ded13c9e6c7ded78cb4 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5198500164)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4830/merge
Commit: c53721e2b81dfb32c96173ba35b360d4257390c9 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5199065513)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4830/merge
Commit: e456e8995f21ea5c7726702d2e2529858d66e5b2 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5199248359)

**Review by mativm02 — APPROVED** [review]
lgtm 🚀

**Review by sredxny — APPROVED** [review]

**Pull request merged**